### ISSUE TITLE
afterDisconnect on sails.lower does not wait

### ISSUE THREAD

**Issue by cschwaderer** [balderdashy/sails#4554]
**Sails version**: 1.1.0-3
**Node version**: 10.11.0
**NPM version**: 6.4.1
**DB adapter name**: N/A
**DB adapter version**: N/A
**Operating system**: Kubuntu 18.04

<hr/>

In `afterDisconnect`, I need to do some asynchronous work. That works for "normal" cases, e.g. the user closes their browser window, socket closes, `afterDisconnect` called.
So far, so good.
However, also I also need that asynchronous logic to be done for each socket when Sails is being shut down. But that doesn't work.

When I close Sails (via Ctr+C) and thus trigger `sails.lower()`, the asynchronous part of the code might not be executed.
Consider this code:

```
afterDisconnect: function(session, socket, callback) {
    console.log("afterDisconnect called");
    setTimeout(function () {
      console.log("Waited long enough!");
            return callback();
    },6000);
  },
```
When triggered by `sails.lower()` I will never see "Waited long enough!" on the console. Same goes if I wrap that in an `async/await` structure:

```
waitForSomeTime: function () {
    console.log("Start to wait!");
    return new Promise(function (resolve,reject) {
      setTimeout(function () {
        console.log("Waited enough!");
        resolve();
      },6000);
    });
  },

  afterDisconnect: async function(session, socket, callback) {
    console.log("afterDisconnect called");
    await module.exports.sockets.waitForSomeTime();
    return callback();
  },
```

I don't know whether `afterDisconnect` was never meant to be fully executed upon `sails.lower()` or if this was un-intentionally done when fixing https://github.com/balderdashy/sails/issues/3397. Anyway, it definitely causes inconvenience for me. Is there at least a work around?

**Comment by sailsbot** [comment]
@cschwaderer Thanks for posting, we'll take a look as soon as possible.

---

For help with questions about Sails, [click here](http://sailsjs.com/support).  If you&rsquo;re interested in hiring @sailsbot and her minions in Austin, [click here](http://sailsjs.com/studio).

**Comment by johnabrams7** [comment]
@cschwaderer Perhaps giving Sails more time before shutting down might help? 

```
sails.on('lower', function() { 
   setTimeout(
      function() { process.exit(1); },
      6000 // 6 seconds or change to the amount of time you need.
   );
});
```

add to the `config/bootstrap.js`

**Comment by cschwaderer** [comment]
Thanks! However, I've found a different, probably better solution

In the callback of `sails.lift` I've added:

```
process.removeAllListeners('SIGINT');

process.on('SIGINT', async function () {
    
    // Do my stuff for every socket with await

    sails.lower(function (err) {

     });
});
```

That way, I can do whatever I want on `SIGINT` and only call `sails.lower` when I'm done. I don't have to rely on a timeout.

However, I think the fact that `sails.lower` does trigger some events (like `afterDisconnect`), but does not wait for their handlers to complete is worth mentioning in the docs - because it's not what I expect.
